### PR TITLE
Fix DB index error on first launch

### PR DESCRIPTION
**Pull request by thellmund**
This fixes the following issue(s):
On the first launch of the app, we accidentally build the database twice. This results in a SQLite exception, as we attempt to create a unique index a second time. See #878 for console output.

This change fixes this problem by ensuring that we only build the database once.

### CODE REVIEW

File: app/src/main/java/de/tum/in/tumcampusapp/database/TcaDb.java [hunk 1]
```diff
@@ -106,7 +106,7 @@
     private static TcaDb instance;
 
     public static synchronized TcaDb getInstance(Context context) {
-        if (instance == null || !instance.isOpen()) {
+        if (instance == null) {
```
Comment (pfent) [inline app/src/main/java/de/tum/in/tumcampusapp/database/TcaDb.java]: The fix is not that easy...
The `|| !instance.isOpen()` was there for a reason, however I don't exactly recall why.
@psukys introduced it with #584, do you maybe recall why this was important?

**Comment by pfent** [comment]
[Travis reports many tests as broken](https://travis-ci.org/TCA-Team/TumCampusApp/builds/390367445#L3335)

**Comment by pfent** [comment]
Do two separate methods `getInstance()` and `create()` make sense here?
Then the tests could create new instances as they like, but on real devices we only have one instance...

**Comment by thellmund** [comment]
There’s a larger problem with the test cases – they fail for all pull requests (despite being displayed as passed on GitHub).

**Comment by kordianbruck** [comment]
@pfent @thellmund feel free to fix the tests. I could for the love of god not figure out why room would be throwing "Illegal connection pointer" during the tests. I've spent like 3 hours on this, so if you have ideas go right ahead. For the mean time I've ignored failing tests.

File: app/src/main/java/de/tum/in/tumcampusapp/database/TcaDb.java (hunk 1 above)
Comment (kordianbruck) [reply app/src/main/java/de/tum/in/tumcampusapp/database/TcaDb.java]: For the test cases. After each case, the DB is closed to force a flush of all cursors. This somehow does no longer work.

**Comment by pfent** [comment]
@kordianbruck that Roboelectric testing framework is a mess, implementing their own SQLite engine, with *special* behaviour...
I'm not toughing that :see_no_evil:

**Comment by kordianbruck** [comment]
@pfent so back to instrumentation tests, now that they are supposed to be faster?

**Comment by pfent** [comment]
Last I checked, it was impossible to run instrumentation tests on travis or on our build server, due to the emulator needing massive resources... :man_shrugging: 
IMHO, we could spend time much more effectively actually working on features, that fighting testing frameworks

**Comment by kordianbruck** [comment]
Y is testing android apps such a mess?

**Pull request merged**